motion_planning/include/motion_planning/planner_client.hpp

```diff
@@ -51,23 +63,22 @@ class planner_client_t
       ROS_WARN("Waiting for observation");
       ros::Duration(0.1).sleep();
     }
-    ROS_DEBUG("Calling planner service");
     _preprocess_start_time = ros::Time::now().toSec();
     _service.request.current_observation = _most_recent_observation;
-    ROS_DEBUG("Current observation: %f, %f, %f, %f, %f, %f", _service.request.current_observation.pose.position.x,
-             _service.request.current_observation.pose.position.y,
-             _service.request.current_observation.pose.orientation.w,
-             _service.request.current_observation.pose.orientation.x,
-             _service.request.current_observation.pose.orientation.y,
-             _service.request.current_observation.pose.orientation.z);
     _service.request.planning_duration.data = ros::Duration(planning_duration);
     _service.request.first_cycle.data = first_cycle;
     _service.request.goal_configuration = goal_configuration;
     _service.request.return_trajectory.data = _publish_trajectory;
     if (_service_client.call(_service))
     {
       ROS_DEBUG("Service call successful");
-      if (_service.response.planner_output == Service::Response::TYPE_SUCCESS)
+      if (is_goal_reached(goal_configuration, goal_radius))
+      {
+        ROS_INFO("Goal reached. Not publishing plan");
+        // TODO: Check if zero control has been published last
```

Comment:
I think having zero control is not the based idea (at some point we talked about having a controller like lqr). Maybe at this point we call a controller, which for now can be a "set zero" one.

Reply:
Will get to this as part of a future PR.